File: TelerikUIExamplesInCSharp/Infrastructure/ViewController.cs
using System;
using System.Drawing;
using System.Collections.Generic;

using Foundation;
using UIKit;

namespace Examples
{
	public partial class ViewController : UIViewController
	{
		UITableView tableView;
		TableViewDataSource tableViewDataSource;
		TableViewDelegate tableViewDelegate;
		public ExampleInfo[] Examples;

		public ViewController(ExampleInfo[] examples)
		{
			this.Examples = examples;
		}
			
		#region View lifecycle

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			if (this.Examples == null) {
				ExampleInfo[] examples = new ExampleInfo[] { 

					new ExampleInfo("Alert", new ExampleInfo[] {
						new ExampleInfo("Getting Started", typeof(AlertGettingStarted)),
						new ExampleInfo("Notifications", typeof(AlertNotifications)),
						new ExampleInfo("Animations", typeof(AlertAnimations)),
						new ExampleInfo("Custom View", typeof(AlertCustomView)),
						new ExampleInfo("Customize", typeof(AlertViewCustomize)),
						new ExampleInfo("Settings", typeof(AlertSettings)),
					}),

					new ExampleInfo("AppFeedback", typeof(FeedbackExampleController)),

					new ExampleInfo("AutoCompleteTextView (Beta)", new ExampleInfo[] {
						new ExampleInfo("Getting Started", typeof(AutoCompleteGettingStarted)),
						new ExampleInfo("Customization", typeof(AutoCompleteCustomize)),
						new ExampleInfo("Tokens", typeof(AutoCompleteTokens)),
					}),

					new ExampleInfo("Calendar", new ExampleInfo[] {
						new ExampleInfo("Calendar with events", typeof(CalendarWithEvents)),
						new ExampleInfo("Transition Effects", typeof(CalendarTransitionEffects)),
						new ExampleInfo("Selection", typeof(CalendarSelection)),
						new ExampleInfo("iOS 7 style calendar", typeof(iOS7StyleCalendar)),
						new ExampleInfo("View modes", typeof(CalendarViewModes)),
						new ExampleInfo("Customization", typeof(CalendarCustomization)),
						new ExampleInfo("EventKit data binding", typeof(CalendarEventKitDataBinding)),
						new ExampleInfo("Localized calendar", typeof(LocalizedCalendar)),
						new ExampleInfo("Inline events", typeof(InlineEvents)),
					}),

					new ExampleInfo("Chart", new ExampleInfo[] {

						new ExampleInfo("Chart Types", new ExampleInfo[] {
							new ExampleInfo("Column / Bar chart", typeof(ColumnAndBarChart)),
							new ExampleInfo("Range Column / Bar chart", typeof(RangeColumnBarChart)),
							new ExampleInfo("Line / Area / Spline chart", typeof(LineAreaSpline)),
							new ExampleInfo("Scatter chart", typeof(ScatterChart)),
							new ExampleInfo("Bubble chart", typeof(BubbleChart)),
							new ExampleInfo("Pie chart", typeof(PieDonut)),
							new ExampleInfo("Stacked Column chart", typeof(StackedColumnChart)),
							new ExampleInfo("Stacked Area chart", typeof(StackedAreaChart)),
							new ExampleInfo("Gaps Line / Spline / Area chart", typeof(GapsLineSplineAreaChart)),
							new ExampleInfo("Financial chart", typeof(FinancialChart)),
							new ExampleInfo("Indicators", typeof(IndicatorsChart)),
						}),

						new ExampleInfo("Axis Types", new ExampleInfo[] {
							new ExampleInfo("Numeric axis", typeof(NumericAxis)),
							new ExampleInfo("Categorical axis", typeof(CategoricalAxis)),
							new ExampleInfo("Date/Time axis", typeof(DateTimeAxis)),
							new ExampleInfo("Date/Time category axis", typeof(DateTimeCategoryAxis)),
							new ExampleInfo("Multiple axes", typeof(MultipleAxes)),
							new ExampleInfo("Negative values", typeof(NegativeValues)),
							new ExampleInfo("Logarithmic Axis", typeof(LogarithmicAxis)),
							new ExampleInfo("Custom Axis", typeof(CustomAxis))
						}),

						new ExampleInfo("Animations", new ExampleInfo[] {
							new ExampleInfo("Default animations", typeof(DefaultAnimation)),
							new ExampleInfo("Custom animation - line chart", typeof(CustomAnimationLineChart)),
							new ExampleInfo("Custom animation - area chart", typeof(CustomAnimationAreaChart)),
							new ExampleInfo("Custom animation - pie chart", typeof(CustomAnimationPieChart)),
							new ExampleInfo("UIKit dynamics animation", typeof(UIKitDynamicsAnimation)),
						}),

						new ExampleInfo("Binding", new ExampleInfo[] {
							new ExampleInfo("Bind with data point", typeof(BindingWithDataPoint)),
							new ExampleInfo("Bind with custom object", typeof(BindingWithCustomObject)),
							new ExampleInfo("Bind with delegate", typeof(BindingWithDelegate)),
						}),

						new ExampleInfo("Pan/Zoom", typeof(PanZoom)),
						new ExampleInfo("Customize", typeof(Customize)),

						new ExampleInfo("Annotations", new ExampleInfo[] {
							new ExampleInfo("Band and line annotations", typeof(BandAndLineAnnotations)),
							new ExampleInfo("Balloon annotation", typeof(BalloonAnnotation)),
							new ExampleInfo("Layer annotation", typeof(LayerAnnotation)),
							new ExampleInfo("View annotation", typeof(ViewAnnotation)),
							new ExampleInfo("Cross line annotation", typeof(CrossLineAnnotation)),
							new ExampleInfo("Custom annotation", typeof(CustomAnnotation)),
							new ExampleInfo("Trackball", typeof(Trackball)),
						}),

						new ExampleInfo("Point Labels", new ExampleInfo[] {
							new ExampleInfo("Point Labels", typeof(PointLabels)),
							new ExampleInfo("Custom Point Label", typeof(CustomPointLabels)),
							new ExampleInfo("Custom Label Render",typeof(CustomPointLabelRender))
						}),
						new ExampleInfo("Live data", typeof(LiveData))
					}),

					new ExampleInfo("DataForm (New)", new ExampleInfo[] {
						new ExampleInfo("Getting Started", typeof(DataFormGettingStarted)),
						new ExampleInfo("Read Only", typeof(DataFormReadOnly)),
						new ExampleInfo("Validation", typeof(DataFormValidation)),
						new ExampleInfo("Customization", typeof(DataFormCustomization)),
						new ExampleInfo("Collapsable Groups", typeof(DataFormCollapsibleGroups)),
						new ExampleInfo("Alignment", typeof(DataFormAlignment)),
						new ExampleInfo("JSON Support", typeof(DataFormJSONSupport))
					}),

					new ExampleInfo("DataSource", new ExampleInfo[] {
						new ExampleInfo("Getting started", typeof(DataSourceGettingStarted)),
						new ExampleInfo("Descriptors API", typeof(DataSourceDescriptorsAPI)),
						new ExampleInfo("Bind with UI controls", typeof(DataSourceUIBindings)),
						new ExampleInfo("Consume web service", typeof(DataSourceWithWebService)),
					}),

					new ExampleInfo("ListView", new ExampleInfo[] {
						new ExampleInfo("Getting Started", typeof(ListViewGettingStarted)),
						new ExampleInfo("Swipe cell", typeof(ListViewSwipe)),
						new ExampleInfo("Items reorder", typeof(ListViewReorder)),
						new ExampleInfo("Selection", typeof(ListViewSelection)),
						new ExampleInfo("Grouping", typeof(ListViewGroups)),
						new ExampleInfo("Layouts", typeof(ListViewLayout)),
						new ExampleInfo("Animations", typeof(ListViewAnimations)),
						new ExampleInfo("Load on demand", typeof(ListViewLoadOnDemand)),
						new ExampleInfo("Pull to refresh", typeof(ListViewPullToRefresh)),
					}),

					new ExampleInfo("SideDrawer", new ExampleInfo[] {
						new ExampleInfo("Getting Started", typeof(SideDrawerGettingStarted)),
						new ExampleInfo("Transitions", typeof(SideDrawerTransitions)),
						new ExampleInfo("Positions", typeof(SideDrawerPositions)),
						new ExampleInfo("Custom Content", typeof(SideDrawerCustomContent)),
						new ExampleInfo("Custom Transition", typeof(SideDrawerCustomTransition)),
						new ExampleInfo("Multiple SideDrawers", typeof(MultipleSideDrawers))
					}),

					new ExampleInfo("Gauges (New)", new ExampleInfo[] {
						new ExampleInfo("Getting Started", typeof(GaugesGettingStarted)),
						new ExampleInfo("Customization", typeof(GaugeCustomization)),
						new ExampleInfo("Interaction", typeof(GaugeInteraction)),
						new ExampleInfo("Animations", typeof(GaugeValueAnimation)),
						new ExampleInfo("Scales", typeof(GaugeMultipleScales)),
						new ExampleInfo("Ranges", typeof(GaugeMultipleRanges)),
					}),
				};
				this.Examples = examples;
				this.Title = "Examples";
			}

			this.tableView = new UITableView ();
			this.tableView.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.tableView.RegisterClassForCellReuse (typeof(UITableViewCell), new NSString("cell"));
			this.tableView.Frame = this.View.Bounds;
			this.tableViewDataSource = new TableViewDataSource (this);
			this.tableViewDelegate = new TableViewDelegate (this);
			this.tableView.DataSource = this.tableViewDataSource;
			this.tableView.Delegate = this.tableViewDelegate;
			this.View.AddSubview (this.tableView);
		}

		#endregion

		public class TableViewDataSource: UITableViewDataSource
		{
			ViewController owner;

			public TableViewDataSource(ViewController owner)
			{
				this.owner = owner;
			}

			protected override void Dispose (bool disposing)
			{
				base.Dispose (disposing);
				this.owner = null;
			}

			public override nint RowsInSection (UITableView tableView, nint section)
			{
				return owner.Examples.Length;
			}

			public override UITableViewCell GetCell (UITableView tableView, NSIndexPath indexPath)
			{
				ExampleInfo info = owner.Examples [indexPath.Row];
				UITableViewCell cell = tableView.DequeueReusableCell ("cell");
				cell.TextLabel.Text = info.Title;
				cell.SelectionStyle = UITableViewCellSelectionStyle.None;
				return cell;
			}
		}

		public class TableViewDelegate: UITableViewDelegate
		{
			ViewController owner;

			public TableViewDelegate(ViewController owner)
			{
				this.owner = owner;
			}

			protected override void Dispose (bool disposing)
			{
				base.Dispose (disposing);
				this.owner = null;
			}

			public override void RowSelected (UITableView tableView, NSIndexPath indexPath)
			{
				ExampleInfo info = owner.Examples [indexPath.Row];
				UIViewController viewController = info.CreateViewController ();
				owner.NavigationController.PushViewController (viewController, true);
				GC.Collect (GC.MaxGeneration, GCCollectionMode.Forced);
			}
		}
	}
}

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/ViewController.swift
//
//  ViewController.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
    
    var examples = [ExampleInfo]()
    
    func initWithExample(example:ExampleInfo) {
        self.examples = example.examples
        self.title = example.title
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view.backgroundColor = UIColor.whiteColor()

        if (examples.count == 0) {
            self.initExamples()
            self.title = "Examples"
        }
        
        let table = UITableView(frame: self.view.bounds)
        table.autoresizingMask = [.FlexibleWidth, .FlexibleHeight]
        table.delegate = self
        table.dataSource = self
        table.registerClass(UITableViewCell.self, forCellReuseIdentifier: "cell")
        self.view.addSubview(table)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func initExamples() {
        
        examples.removeAll()

        examples.append(ExampleInfo(title: "Alert", exampleList: [
            ExampleInfo(title: "Getting started") { AlertGettingStarted() },
            ExampleInfo(title: "Animations") { AlertAnimations() },
            ExampleInfo(title: "Notifications") { AlertNotifications() },
            ExampleInfo(title: "Custom View") { AlertCustomView() },
            ExampleInfo(title: "Customize") { AlertCustomize() },
            ExampleInfo(title: "Settings") { AlertSettings() },
            ]))
        
        examples.append(ExampleInfo(title: "AppFeedback") { FeedbackExampleController(nibName: "FeedbackExampleController", bundle: nil) })
        
        examples.append(ExampleInfo(title: "AutoCompleteTextView (Beta)", exampleList:[
            ExampleInfo(title: "Getting Started") { AutoCompleteGettingStartedViewController() },
            ExampleInfo(title: "Customization") { AutoCompleteCustomization() },
            ExampleInfo(title: "Tokens") { AutoCompleteTokens() },
            ]))
        
        examples.append(ExampleInfo(title: "Calendar", exampleList:[
            ExampleInfo(title: "Calendar with events") { CalendarWithEvents() },
            ExampleInfo(title: "View modes") { CalendarViewModes() },
            ExampleInfo(title: "Transition effects") { CalendarTransitonEffects() },
            ExampleInfo(title: "Selection") { CalendarSelection() },
            ExampleInfo(title: "iOS 7 style calendar") { iOS7StyleCalendar() },
            ExampleInfo(title: "Customization") { CalendarCustomization() },
            ExampleInfo(title: "EventKit data binding") { CalendarEventKitDataBinding() },
            ExampleInfo(title: "Localized calendar") { LocalizedCalendar() },
            ExampleInfo(title: "Inline events") { InlineEvents() },
            ]))
        
        examples.append(ExampleInfo(title: "Chart", exampleList:[
            ExampleInfo(title: "Chart Types", exampleList:[
                ExampleInfo(title: "Column / Bar chart") { ColumnAndBarChart() },
                ExampleInfo(title: "Range Column / Bar chart") { RangeBarColumnChart() },
                ExampleInfo(title: "Line / Area / Spline chart") { LineAreaSpline() },
                ExampleInfo(title: "Scatter chart") { ScatterChart() },
                ExampleInfo(title: "Bubble chart") { BubbleChart() },
                ExampleInfo(title: "Pie chart") { PieDonut() },
                ExampleInfo(title: "Stacked Column chart") { StackedColumnChart() },
                ExampleInfo(title: "Stacked Area chart") { StackedAreaChart() },
                ExampleInfo(title: "Gaps Line / Spline / Area chart") { GapsLineSplineAreaChart() },
                ExampleInfo(title: "Financial chart") { FinancialChart() },
                ExampleInfo(title: "Indicators") { Indicators() },
                ]),
            ExampleInfo(title: "Axis Types", exampleList:[
                ExampleInfo(title: "Numeric axis") { NumericAxis() },
                ExampleInfo(title: "Categorical axis") { CategoricalAxis() },
                ExampleInfo(title: "Date/Time axis") { DateTimeAxis() },
                ExampleInfo(title:"Date/Time category axis") { DateTimeCategoryAxis() },
                ExampleInfo(title: "Multiple axes") { MultipleAxes() },
                ExampleInfo(title: "Negative values") { NegativeValues() },
                ExampleInfo(title: "Logarithmic axis") { LogarithmicAxis() },
                ExampleInfo(title: "CustomAxis"){ CustomAxis() }
                ]),
            ExampleInfo(title: "Animations", exampleList:[
                ExampleInfo(title: "Default animations") { DefaultAnimation() },
                ExampleInfo(title: "Custom animation - line chart") { CustomAnimationLineChart() },
                ExampleInfo(title: "Custom animation - area chart") { CustomAnimationAreaChart() },
                ExampleInfo(title: "Custom animation - pie chart") { CustomAnimationPieChart() },
                ExampleInfo(title: "UIKit dynamics animation") { UIKitDynamicsAnimation() }
                ]),
            ExampleInfo(title: "Binding", exampleList:[
                ExampleInfo(title: "Bind with data point") { BindWithDataPoint() },
                ExampleInfo(title: "Bind with custom object") { BindWithCustomObject() },
                ExampleInfo(title: "Bind with delegate") { BindWithDelegate() }
                ]),
            ExampleInfo(title: "Pan/Zoom") { PanZoom() },
            ExampleInfo(title: "Customize") { Customize() },
            ExampleInfo(title: "Annotations", exampleList:[
                ExampleInfo(title: "Band and line annotations") { BandAndLineAnnotation() },
                ExampleInfo(title: "Balloon annotation") { BalloonAnnotation() },
                ExampleInfo(title: "Layer annotation") { LayerAnnotation() },
                ExampleInfo(title: "View annotation") { ViewAnnotation() },
                ExampleInfo(title: "Cross line annotation") { CrossLineAnnotation() },
                ExampleInfo(title: "Custom annotation") { CustomAnnotation() },
                ExampleInfo(title: "Trackball") { Trackball() }
                ]),
            ExampleInfo(title: "Point Labels", exampleList: [
                ExampleInfo(title: "Point Labels") { PointLabels() },
                ExampleInfo(title: "Custom Label") { CustomPointLabels() },
                ExampleInfo(title: "Custom Label Render") { CustomPointLabelRender() }
                ]),
            ExampleInfo(title: "Live data") { LiveData() }
            ]))
        
        examples.append(ExampleInfo(title: "DataForm (New)", exampleList: [
            ExampleInfo(title: "Getting started") { DataFormGettingStarted()},
            ExampleInfo(title: "Validation") { DataFormValidation()},
            ExampleInfo(title: "Read Only") { DataFormReadOnly()},
            ExampleInfo(title: "Customization") { DataFormCustomization()},
            ExampleInfo(title: "Collapsible Groups") { DataFormCollapsibleGroups()},
            ExampleInfo(title: "Alignment") { DataFormLabelsAlignment()},
            ExampleInfo(title: "JSON Support") { DataFormJSONSupport()}
            ]))
        
        examples.append(ExampleInfo(title: "DataSource", exampleList:[
            ExampleInfo(title: "Getting started") { DataSourceGettingStarted() },
            ExampleInfo(title: "Descriptors API") { DataSourceDescriptorsAPI() },
            ExampleInfo(title: "Bind with UI controls") { DataSourceUIBindings() },
            ExampleInfo(title: "Consume web service") { DataSourceWithWebService() },
            ]))
        
        examples.append(ExampleInfo(title: "Gauges (New)", exampleList: [
            ExampleInfo(title: "Getting started") { GaugeGettingStarted() },
            ExampleInfo(title: "Customization") { GaugeCustomization() },
            ExampleInfo(title: "Interaction") { GaugeInteraction() },
            ExampleInfo(title: "Animations") { GaugeAnimations() },
            ExampleInfo(title: "Scales") { GaugeScales() },
            ExampleInfo(title: "Ranges") { GaugeRanges() },
            ]))
        
        examples.append(ExampleInfo(title: "ListView", exampleList: [
            ExampleInfo(title: "Getting started") { ListViewGettingStarted() },
            ExampleInfo(title: "Swipe cell") { ListViewSwipe() },
            ExampleInfo(title: "Items reorder") { ListViewReorder() },
            ExampleInfo(title: "Selection") { ListViewSelection() },
            ExampleInfo(title: "Grouping") { ListViewGroups() },
            ExampleInfo(title: "Layouts"){ ListViewLayout() },
            ExampleInfo(title: "Animations") { ListViewAnimations() },
            ExampleInfo(title: "Load on demand") { ListViewLoadOnDemand() },
            ExampleInfo(title: "Pull to refresh") { ListViewPullToRefresh() },
            ]))
        
        examples.append(ExampleInfo(title: "SideDrawer", exampleList: [
            ExampleInfo(title: "Getting Started") { SideDrawerGettingStarted() },
            ExampleInfo(title: "Transitions") { SideDrawerTransitions() },
            ExampleInfo(title: "Custom Content") { SideDrawerCustomContent() },
            ExampleInfo(title: "Custom Transition") { SideDrawerCustomTransition() },
            ExampleInfo(title: "Positions") { SideDrawerPositions() },
            ExampleInfo(title: "Multiple SideDrawers") { MultipleSideDrawers() }
            ]))
    }
    
//MARK: - UITableViewDataSource
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return examples.count
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("cell")!
        cell.selectionStyle = UITableViewCellSelectionStyle.None
        cell.textLabel!.text = examples[indexPath.row].title
        return cell
    }
    
//MARK: - UITableViewDelegate
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        let controller = examples[indexPath.row].createController()
        self.navigationController?.pushViewController(controller, animated: true)
    }
}

